### ISSUE TITLE
Research/identify plugins: barcode scanner, geo location, camera

### ISSUE THREAD

**Issue by yxu-lanl** [microbiomedata/nmdc-field-notes#17]

**Comment by ssarrafan** [comment]
@yxu-lanl I added this issue to the sprint board since you changed the status to in progress.  Let me know if you're not currently working on it.

**Comment by eecavanna** [comment]
Hi @yxu-lanl, is the task here that you want to research/identify plugins for these, or integrate those plugins into the code base? If the latter, I'd recommend splitting this up into multiple issues (so that each plugin is introduced via its own PR).

**Comment by yxu-lanl** [comment]
I want to research/identify plugins for these.

**Comment by eecavanna** [comment]
OK, thanks. 👍

By the way, I looked into barcode-reading functionality (only) a few weeks ago and found this package.

`@capacitor-mlkit/barcode-scanning` (Apache 2.0 license; supports QR codes and other kinds of barcodes)

I don't have any experience with it, though.

**Comment by yxu-lanl** [comment]
Thanks. I found another one with ionic/react typescript code example. https://github.com/danielsogl/awesome-cordova-plugins

**Comment by ssarrafan** [comment]
Moving to next sprint, Yan is working on this

**Comment by yxu-lanl** [comment]
import { Barcode, BarcodeScanner } from '@capacitor-mlkit/barcode-scanning';
import { Geolocation, Position } from '@capacitor/geolocation';
import { Camera, CameraResultType, Photo } from '@capacitor/camera';

![Screenshot 2024-02-12 at 11 47 26 AM](https://github.com/microbiomedata/nmdc-field-notes/assets/38266469/989a0b2b-2a81-4305-a2fa-91e95414ddde)

**Comment by eecavanna** [comment]
I wonder if it has to do with the API version being used for the emulator. 

I looked up "API level 28" and found [this list of API versions](https://developer.android.com/tools/releases/platforms#expandable-4). Based on that list, I think API level 28 was released in 2018 alongside Android 9. Here are some of the more recent versions:
- Android 10 (API level 29)
- Android 11 (API level 30)
- Android 12 (API levels 31 and 32)
- Android 13 (API level 33)
- Android 14 (API level 34)

I wonder if the documentation for the `@capacitor-mlkit/barcode-scanning` library shows a minimum API requirement.
